helix-core/src/main/java/org/apache/helix/task/TaskStateModel.java

```diff
@@ -43,6 +49,11 @@
   private ScheduledFuture timeout_task;
   private TaskRunner _taskRunner;
   private final ScheduledExecutorService _timeoutTaskExecutor;
+  public static final String TASK_JAR_FILE = "JAR_FILE";
+  public static final String TASK_VERSION = "VERSION";
+  public static final String TASK_CLASSES = "TASK_CLASSES";
+  public static final String TASK_FACTORY = "TASKFACTORY";
+  public static final String TASK_PATH = "TASK_DEFINITION";
```

Comment:
Overall, the code still feels like a "script" - to make this more amenable for use in production, let's work on making it maintainable and extendable :)